osquery/logger/plugins/aws_util.cpp

```diff
@@ -356,4 +357,43 @@ Status getAWSRegion(std::string& region, bool sts) {
   VLOG(1) << "Using default AWS region: " << region;
   return Status(0);
 }
+
+Status appendLogTypeToJson(const std::string& log_type,
+                           const std::string& original,
+                           std::string& serialized) {
+  if (log_type.empty()) {
+    return Status(1, "log_type is empty");
+  }
+
+  if (original.empty()) {
+    return Status(1, "original JSON is empty");
+  }
+
+  pt::ptree params;
+  try {
+    std::stringstream input;
+    input << original;
+    pt::read_json(input, params);
+  } catch (const pt::json_parser::json_parser_error& e) {
+    return Status(1,
+                  std::string("JSON deserialization exception: ") + e.what());
+  }
+
+  params.put<std::string>("log_type", log_type);
```

Comment:
@theopolis It may be possible to add another argument (maybe a `bool` indicating whether to label the log item?) to `logString` and `logStatus` in `BufferedLogForwarder`. On `true`, this would append an extra `log_type` key with `"status"` in `logStatus`. For `logString`, it would deserialize the string as JSON and append the `log_type` key with `"results"` prior to reserialization.